### PR TITLE
Migration: Cypress tests for NamespaceStore

### PR DESCRIPTION
**Pull request by vbnrh**
Screenshot from 2022-01-18 17-06-48

Signed-off-by: Vineet Badrinath <vineetbnath@gmail.com>

### CODE REVIEW

File: cypress/views/store.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      case Providers.S3:
+        setupS3Endpoint(storeType);
+        break;
+      case Providers.PVC:
+        setupPVC();
+        break;
+      default:
+        break;
+    }
+  };
```
[the remark this answers is not in the document]
Comment (vbnrh) [reply cypress/views/store.ts]: great, thanks !

File: cypress/views/store.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    inputCustomSecrets(storeType);
+};
+  
+  export const createStore = (provider: Providers, storeType: StoreType = StoreType.BackingStore) => {
+    cy.log(`Creating ${storeType}`);
+    cy.byTestID(`${storeType}-name`).type(testName);
+    setupProvider(provider, storeType);
+    cy.byTestID(`${storeType}-create-button`).click();
+  };
+
```
Comment (bipuladh) [inline cypress/views/store.ts:49]: New line needed.

File: cypress/tests/namespace-store.spec.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      });
+      
+    beforeEach(() => {
+      cy.visit('/');
+      ODFCommon.visitStorageDashboard();
+      cy.byLegacyTestID('horizontal-link-Namespace Store').click();
+      cy.byTestID('item-create').click();
+    });
+  
+    it('Test creation of AWS namespace store', () => {
```
Comment (bipuladh) [inline cypress/tests/namespace-store.spec.ts:31]: If we are creating secret as part of the UI then please check for the existence of secret as well. If Noobaa handles secret creation then not required.

**Review by bipuladh — CHANGES_REQUESTED** [review]

File: cypress/views/store.ts (hunk 2 above)
Comment (vbnrh) [reply cypress/views/store.ts:50]: done

File: cypress/tests/namespace-store.spec.ts (hunk 3 above)
Comment (vbnrh) [reply cypress/tests/namespace-store.spec.ts:36]: done

File: cypress/tests/namespace-store.spec.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  
+    const checkSecret = (tName: string) => {
+      cy.exec(`oc get secrets ${tName}-secret -n openshift-storage`, {failOnNonZeroExit: true, log: true, timeout: 2 * MIN})
+      .its('code').
+      should('eq', 0);
+    } 
+    it('Test creation of AWS namespace store', () => {
+      createStore(Providers.AWS, StoreType.NamespaceStore);
+      cy.byLegacyTestID('resource-title').contains(testName);
+      checkSecret(testName)
```
Comment (bipuladh) [inline cypress/tests/namespace-store.spec.ts:39]:
```suggestion
      checkSecret(testName);
```

File: cypress/tests/namespace-store.spec.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    });
+  
+    it('Test creation of Azure namespace store', () => {
+      createStore(Providers.AZURE, StoreType.NamespaceStore);
+      cy.byLegacyTestID('resource-title').contains(testName);
+      checkSecret(testName)
```
Comment (bipuladh) [inline cypress/tests/namespace-store.spec.ts:45]:
```suggestion
      checkSecret(testName);
```

File: cypress/tests/namespace-store.spec.ts [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    });
+  
+    it('Test creation of S3 Endpoint Type', () => {
+      createStore(Providers.S3, StoreType.NamespaceStore);
+      cy.byLegacyTestID('resource-title').contains(testName);
+      checkSecret(testName)
```
Comment (bipuladh) [inline cypress/tests/namespace-store.spec.ts:51]:
```suggestion
      checkSecret(testName);
```

File: cypress/views/odf-common.ts [hunk 7]
```diff
@@ -7,3 +7,5 @@ export const ODFCommon = {
         cy.contains('Storage Systems').click();
     }
 }
+
+export const MIN = 60000
```
Comment (bipuladh) [inline cypress/views/odf-common.ts:11]: new line

File: cypress/views/store.ts (hunk 2 above)
Comment (bipuladh) [reply cypress/views/store.ts:50]: still see the red icon on gh

File: cypress/tests/namespace-store.spec.ts [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+      ODFCommon.visitStorageDashboard();
+      cy.byLegacyTestID('horizontal-link-Namespace Store').click();
+      cy.byTestID('item-create').click();
+    });
+  
+    const checkSecret = (tName: string) => {
+      cy.exec(`oc get secrets ${tName}-secret -n openshift-storage`, {failOnNonZeroExit: true, log: true, timeout: 2 * MIN})
+      .its('code').
+      should('eq', 0);
+    }
```
Comment (bipuladh) [inline cypress/tests/namespace-store.spec.ts:35]: Please run prettier on these lines

**Review by bipuladh — CHANGES_REQUESTED** [review]

**Review by bipuladh — APPROVED** [review]
/approve
/lgtm

**Pull request merged**